plugin/plugin.go

```diff
@@ -104,46 +77,67 @@ func (e *Ratelimit) Evaluate(r rules.RuleMetadata, tx rules.TransactionState) {
 
 	corazaLogger.Debug().Msg("Evaluating ratelimit plugin")
 
+	var request_allowed bool
+
 	//extract zone
-	zone_name := e.zoneMacro.Expand(tx)
-	if zone_name == "" {
-		zone_name = "misc" // if in case of empty string or some kind of issue in macro expansion we send all the requests to misc name
-	}
 
-	currentTimeInSecond := time.Now().Unix()
+	// MultiZones behave in 'OR' manner, REQUEST will be allowed if any one of the zone is allowing the request
 
 	e.mutex.Lock()
```

Comment:
since there is an iteration later I would move the lock down and do per iteration. Also work out the values like https://github.com/VermaShivansh/coraza-ratelimit-plugin/pull/8/files#diff-f42d64c75c9ca7a08c149ea609687764f2abcd3a79ae4e77110c6ae44ba36e9fR90-R94 out of the lock. Remember the more expensive the operation is, the more this becomes a bottleneck.